// Copyright 2019 Zeppelin Bend Pty Ltd
// This file is part of cimbend.
// 
// cimbend is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
// 
// cimbend is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
// 
// You should have received a copy of the GNU Affero General Public License
// along with cimbend.  If not, see <https://www.gnu.org/licenses/>.


syntax = "proto3";
option java_multiple_files = true;
option java_package = "com.zepben.protobuf.cim.iec61970.base.core";
package zepben.cim.iec61970.base.core;

/**
 * An unordered enumeration of phase identifiers.  Allows designation of phases for both transmission and distribution equipment,
 * circuits and loads.   The enumeration, by itself, does not describe how the phases are connected together or connected to ground.
 * Ground is not explicitly denoted as a phase.
 * <p>
 * Residential and small commercial loads are often served from single-phase, or split-phase, secondary circuits. For example of s12N,
 * phases 1 and 2 refer to hot wires that are 180 degrees out of phase, while N refers to the neutral wire. Through single-phase
 * transformer connections, these secondary circuits may be served from one or two of the primary phases A, B, and C. For three-phase
 * loads, use the A, B, C phase codes instead of s12N.
 */
enum PhaseCode {
    /**
     * No phases specified.
     */
    NONE = 0;
    /**
     * Phase A.
     */
    A = 1;
    /**
     * Phase B.
     */
    B = 2;
    /**
     * Phase C.
     */
    C = 3;
    /**
     * Neutral phase.
     */
    N = 4;
    /**
     * Phases A and B.
     */
    AB = 5;
    /**
     * Phases A and C.
     */
    AC = 6;
    /**
     * Phases A and neutral.
     */
    AN = 7;
    /**
     * Phases B and C.
     */
    BC = 8;
    /**
     * Phases B and neutral.
     */
    BN = 9;
    /**
     * Phases C and neutral.
     */
    CN = 10;
    /**
     * Phases A, B, and C.
     */
    ABC = 11;
    /**
     * Phases A, B, and neutral.
     */
    ABN = 12;
    /**
     * Phases A, C and neutral.
     */
    ACN = 13;
    /**
     * Phases B, C, and neutral.
     */
    BCN = 14;
    /**
     * Phases A, B, C, and N.
     */
    ABCN = 15;
    /**
     * Unknown non-neutral phase.
     */
    X = 16;
    /**
     * Unknown non-neutral phase plus neutral.
     */
    XN = 17;
    /**
     * Two unknown non-neutral phases.
     */
    XY = 18;
    /**
     * Two unknown non-neutral phases plus neutral.
     */
    XYN = 19;
}
